### ISSUE TITLE
Typescript errors are un-dismissable if you click them

### ISSUE THREAD

**Issue by msfeldstein** [zed-industries/zed#4819]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Clicking the hovering typescript error causes it to show up below the line, rather than as a popup above, and once its there there's no way to get rid of it.

![image](https://user-images.githubusercontent.com/161135/230453524-0c990122-031c-4e84-bbef-73771f038d7b.png)

https://user-images.githubusercontent.com/161135/230453398-088a3654-f303-421b-8dbd-2a10a7d8f4ef.mov

### Environment

Zed: v0.80.5 (stable)
OS: macOS 13.3.0
Memory: 8 GiB
Architecture: aarch64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

_No response_

**Comment by nilskch** [comment]
> At the moment, esc should dismiss the error - a few others have found this a bit hard to discover and not super intuitive - not exactly sure what would be best for us to do.

It would be nice if the diagnostics would disappear once the error is resolved (as suggested in #8397). Currently, I manually hit `ESC` after resolving the error as you said. I also found this hard to discover; I opened and closed the file a few times until I actively searched for the `ESC` solution. 

> I'm curious what the purpose of the secondary (post-click persistent) state is, as it doesn't give any more info than the hover error.

Sometimes there is additional information to the tooltip and sometimes there is not. In my opinion, it would be nice if the diagnostics only open if there is additional information to display (just my opinion, I would understand if you see this differently).

One more (thing that is not completely related to the issue): depending on the font size and theme, the text in the diagnostics looks very much like the code itself. The tooltip (the box when you hover the error) looks very nice! I think the diagnostics could use a little more styling. Maybe also a box? Not sure here.

<img width="628" alt="Screenshot 2024-02-27 at 22 09 21" src="https://github.com/zed-industries/zed/assets/52418412/bda8942c-4cec-4ad0-902d-2a1c18a8963f">


This is also just my opinion. I found it a little bit confusing at times and I like the box more.
